sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/Changefeed.java

```diff
@@ -0,0 +1,144 @@
+ * The listing of the $blobchangefeed/idx/segments/ virtual directory shows these segments ordered by time.
+ * The path of the segment describes the start of the hourly time-range that the segment represents.
+ * This list can be used to filter out the segments of logs that are interest.
+ *
+ * Note: The time represented by the segment is approximate with bounds of 15 minutes. So to ensure consumption of
+ * all records within a specified time, consume the consecutive previous and next hour segment.
+ */
+class Changefeed {
+
+    private final ClientLogger logger = new ClientLogger(Changefeed.class);
+
+    private static final String SEGMENT_PREFIX = "idx/segments/";
+    private static final String METADATA_SEGMENT_PATH = "meta/segments.json";
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final BlobContainerAsyncClient client; /* Changefeed container */
+    private final OffsetDateTime startTime; /* User provided start time. */
+    private final OffsetDateTime endTime; /* User provided end time. */
+    private OffsetDateTime lastConsumable; /* Last consumable time. The latest time the changefeed can safely be
+                                              read from.*/
+    private OffsetDateTime safeEndTime; /* Soonest time between lastConsumable and endTime. */
+    private final ChangefeedCursor cfCursor; /* Cursor associated with changefeed. */
+    private final ChangefeedCursor userCursor; /* User provided cursor. */
+    private final SegmentFactory segmentFactory; /* Segment factory. */
+
+    /**
+     * Creates a new Changefeed.
+     */
+    Changefeed(BlobContainerAsyncClient client, OffsetDateTime startTime, OffsetDateTime endTime,
+        ChangefeedCursor userCursor, SegmentFactory segmentFactory) {
+        this.client = client;
+        this.startTime = startTime;
+        this.endTime = endTime;
+        this.userCursor = userCursor;
+        this.segmentFactory = segmentFactory;
+
+        this.cfCursor = new ChangefeedCursor(this.endTime);
+        this.safeEndTime = endTime;
+    }
+
+    /**
+     * Get all the events for the Changefeed.
+     * @return A reactive stream of {@link BlobChangefeedEventWrapper}
+     */
+    Flux<BlobChangefeedEventWrapper> getEvents() {
+        return validateChangefeed()
+            .then(populateLastConsumable())
+            .thenMany(listYears())
+            .concatMap(this::listSegmentsForYear)
+            .concatMap(this::getEventsForSegment);
+    }
+
+    /**
+     * Validates that changefeed has been enabled for the account.
+     */
+    private Mono<Boolean> validateChangefeed() {
+        return this.client.exists()
+            .flatMap(exists -> {
+                if (exists == null || !exists) {
+                    return FluxUtil.monoError(logger, new RuntimeException("Changefeed has not been enabled for "
+                        + "this account."));
```

Comment:
is it the case that when change feed is enabled then storage account will have some special container to store them? so its absence means change-feed-not-enabled?

Reply:
Yes, so the changefeed feature relies on the existence of this special container that the service creates. All the blobs we read for this feature are in this container.